### PR TITLE
Support numeric strings for number type

### PR DESCRIPTION
**Pull request by JohnyDays**

### CODE REVIEW

File: lib/document.js [hunk 1]
```diff
@@ -455,6 +455,14 @@ Document.prototype.__makeSavableCopy = function(doc, schema, options, model, r)
       return doc;
     }
   }
+  else if (type.isNumber(schema) && (typeof doc === 'string')) {
+    var numericString = parseInt(doc, 10);
```
Comment (neumino) [inline lib/document.js]: s/parseInt/parseFloat

File: lib/type/number.js [hunk 2]
```diff
@@ -109,6 +109,13 @@ TypeNumber.prototype.validate = function(number, prefix, options) {
     throw new Errors.ValidationError("Validator for the field "+prefix+" returned `false`.");
   }
 
+  if(typeof number === 'string'){
+    var numericString = parseInt(number, 10);
```
Comment (neumino) [inline lib/type/number.js]: Same thing
s/parseInt/parseFloat

**Comment by neumino** [comment]
Same thing here, it should be parseFloat and not parseInt

File: lib/type/number.js [hunk 3]
```diff
@@ -109,6 +109,13 @@ TypeNumber.prototype.validate = function(number, prefix, options) {
     throw new Errors.ValidationError("Validator for the field "+prefix+" returned `false`.");
   }
 
+  if(typeof number === 'string'){
```
Comment (neumino) [inline lib/type/number.js]: Can you merge the `if` with the rest of the `if`/`else if` statement

**Comment by neumino** [comment]
Can you also add a test for saving and a test for parsing to make sure that it works as intended?

**Comment by JohnyDays** [comment]
Done, thanks!
Amazing test suite btw

**Comment by JohnyDays** [comment]
Giving it another look, I didn't merge the ifs because we need the converted number to go through the same checks after it's conversion. Am I missing something or?

**Pull request merged**